jobsdb/jobsdb_import.go

```diff
@@ -23,42 +22,23 @@ func (jd *HandleT) getDsForImport(dsList []dataSetT) dataSetT {
 	return ds
 }
 
-func (jd *HandleT) setupSequenceProvider() {
+//GetLastJobIDBeforeImport should return the largest job id stored so far
+func (jd *HandleT) GetLastJobIDBeforeImport() int64 {
 	jd.assert(jd.migrationState.dsForNewEvents.Index != "", "dsForNewEvents must be setup before calling this")
 	jd.dsListLock.Lock()
 	defer jd.dsListLock.Unlock()
 
-	if jd.migrationState.sequenceProvider == nil || !jd.migrationState.sequenceProvider.IsInitialized() {
-		dsList := jd.getDSList(true)
-
-		var lastJobIDBeforeNewImports int64
-		if lastJobIDBeforeNewImports == 0 {
-			for idx, dataSet := range dsList {
-				if dataSet.Index == jd.migrationState.dsForNewEvents.Index {
-					if idx > 0 {
-						lastJobIDBeforeNewImports = jd.getMaxIDForDs(dsList[idx-1])
-					}
-				}
+	dsList := jd.getDSList(true)
+	var lastJobIDBeforeNewImports int64
+	for idx, dataSet := range dsList {
+		if dataSet.Index == jd.migrationState.dsForNewEvents.Index {
+			if idx > 0 {
+				lastJobIDBeforeNewImports = jd.getMaxIDForDs(dsList[idx-1])
```

Comment:
How do we guarantee that new writes are not happening in this DS? Can we assert it somehow?